stubs/docutils/docutils/io.pyi

```diff
@@ -23,6 +23,18 @@ def error_string(err: BaseException) -> str: ...
 class Input(TransformSpec):
     component_type: ClassVar[str]
     default_source_path: ClassVar[str | None]
+    def __init__(
+        self,
+        source: str | None = None,
```

Comment:
Looking at the sub-classes, source seems to be some kind of file-like type. Using a protocol would be best, although using `IO` would do in a pinch.

Reply:
changed to `source: IO[str] | None = None`. Is that what you had in mind?